packer/builders/amazon-ebs.json

```diff
@@ -14,6 +14,8 @@
   "ami_virtualization_type"  : "hvm",
   "ami_users"  : "{{user `ami_users`}}",
   "enhanced_networking"  : "true",
+  "associate_public_ip_address": "true",
```

Comment:
Mmm, this should be specified under the variables of the instance-template. 
Reasons is that sometimes you may not want to have a public_ip address associated, especially in building a machine.
Did you try with false?